screen/Running/RunningScreen.js

```diff
@@ -105,110 +105,120 @@ const RunningScreen = ({ route, navigation }) => {
+    setHasGameStarted(false);
     setHasOptionClicked(false);
   };
 
   const handlePressOptionButton = () => {
-    pauseGameStatus();
+    gameController.pauseGameStatus();
+    setHasGameStarted(false);
     setHasOptionClicked(true);
   };
 
   const handleFinishDistanceResult = () => {
     setHasGameFinished(true);
   };
 
-  const handleFinishGame = (passedTime) => {
+  const handleFinishGame = (passedTime, survivorCount) => {
+    if (mode === 'survival') {
+      survivalTime.current = passedTime;
+
+      if (survivorCount === 1) {
+        setIsWinner(true);
+      }
+    }
+
     if (passedTime === 0) {
-      setIsWinner(true);
+      if (role === 'human') {
+        setIsWinner(true);
+      }
     } else {
       survivalTime.current -= passedTime;
     }
 
     setHasGameFinished(true);
   };
 
-  const headerOptionButton = useCallback(() => {
-    return (
-      <FontAwesome
-        name="gear"
-        style={styles.option}
-        onPress={handlePressOptionButton}
-      />
-    );
-  }, []);
-
-  useEffect(() => {
-    const setNavigatorOptions = () => {
-      navigation.setOptions({
-        headerTitle: '',
-        headerLeft: () => {},
-        headerRight: headerOptionButton,
-      });
-    };
-
-    setNavigatorOptions();
-  }, [navigation]);
-
   useEffect(() => {
     const initStartUp = () => {
       getCurrentLocation();
 
-      audioController.loadAudio();
-      countDown.current = setTimeout(startRunning, 5000);
+      gameController.loadGameSound();
+      gameController.timeoutId.current = setTimeout(startRunning, 5000);
     };
 
     initStartUp();
 
     return () => {
-      clearTimeout(countDown.current);
-      tracker.current?.remove();
-      audioController.resetAudio();
+      gameController.resetGameSetup('timeout');
     };
   }, []);
 
   useEffect(() => {
-    const startZombieChasing = () => {
-      intervalId.current = setInterval(() => {
-        setZombieDistance((previousDistance) => {
-          const reducedZombieDistance = previousDistance + speedMeterPerSecond;
+    const startOpponentRunning = () => {
+      if (mode === 'oneOnOne') {
+        socket.on('game/opponentSpeed', (meterPerSecond) => {
+          setOpponentDistance((previousDistance) => {
+            const reducedOpponentDistance = previousDistance + meterPerSecond;
+
+            return reducedOpponentDistance;
+          });
+        });
+
+        return;
+      }
 
-          return reducedZombieDistance;
+      if (mode === 'survival') {
+        socket.on('game/die', () => {
+          setUserCount((prev) => prev - 1);
         });
-      }, 1000);
+      }
+
+      if (mode === 'solo' || mode === 'survival') {
+        gameController.intervalId.current = setInterval(() => {
+          setOpponentDistance((previousDistance) => {
+            const reducedOpponentDistance =
+              previousDistance + speedMeterPerSecond;
+            return reducedOpponentDistance;
+          });
+        }, 1000);
```

Comment:
의미를 명확히 하기 위해 변수명을 길게 쓰셨다고 생각됩니다. 개인적으로는 모든 변수가 길게 느껴지면 오히려 코드를 읽기가 힘들 수도 있다는 생각이 듭니다. 저는 이런 경우 아래와 같이 줄여쓰는 것을 선호합니다.

```suggestion
        gameController.intervalId.current = setInterval(() => {
          setOpponentDistance((prev) => {
            const totalDistance = prev + speedMeterPerSecond;
            return totalDistance;
          });
        }, 1000);
```
취향차일 수도 있으니, 참고만 해주세요!

Reply:
계속해서 반복되는 변수명을 활용해주면 가독성이 저해될까봐 좀 더 상세하게 해당 부분을 나누어 주었습니다. 하지만 코드가 더 짧아지는 부분이 더 좋을 수도 있겠네요. 수정하겠습니다.